pkg/apis/controller/v1beta1/appwrapper.go

```diff
@@ -266,11 +267,24 @@ type AppWrapperStatus struct {
 
 	// Represents multi-cluster observatioins
 	TargetClusterName string `json:"targetClusterName"`
+
+	ItemCompletionStatus GenericItemCompletionStatus `json:"itemCompletionStatus,omitempty"`
+}
+
+type GenericItemCompletionStatus struct {
+	GenericItems [] GenericItem `json:"genericItems,omitempty"`
+}
+
+type GenericItem struct {
+	schema.GroupVersionKind `json:"gvk"`
```

Comment:
Can you reach out to Alexey to see if this field works for him.  Consumers of this field will need to match gvk. Would like to get feed back before merge.

Reply:
I reached out to Alexey but he ran out of time to address it today. Will be back on Sunday. Should we wait with merge till Monday?